Modules/Tests/YosemiteTests/PointOfSale/POSSystemStatusServiceTests.swift

```
@@ -0,0 +1,89 @@
+    }
+
+    @Test func loadWooCommercePluginAndPOSFeatureSwitch_returns_plugin_and_enabled_feature_when_feature_is_enabled() async throws {
+        // Given
+        network.simulateResponse(requestUrlSuffix: "system_status", filename: "system-status-wc-plugin-and-pos-feature-enabled")
+
+        // When
+        let result = try await sut.loadWooCommercePluginAndPOSFeatureSwitch(siteID: sampleSiteID)
+
+        // Then
+        let plugin = try #require(result.wcPlugin)
+        #expect(plugin.plugin == "woocommerce/woocommerce.php")
+        #expect(plugin.name == "WooCommerce")
+        #expect(plugin.version == "10.0.0-dev")
+        #expect(plugin.active == true)
+        #expect(result.featureValue == true)
+    }
+
+    @Test func loadWooCommercePluginAndPOSFeatureSwitch_returns_plugin_and_nil_feature_when_feature_is_disabled() async throws {
+        // Given
+        network.simulateResponse(requestUrlSuffix: "system_status", filename: "system-status-wc-plugin-and-pos-feature-disabled")
+
+        // When
+        let result = try await sut.loadWooCommercePluginAndPOSFeatureSwitch(siteID: sampleSiteID)
+
+        // Then
+        let plugin = try #require(result.wcPlugin)
+        #expect(plugin.plugin == "woocommerce/woocommerce.php")
+        #expect(plugin.name == "WooCommerce")
+        #expect(plugin.version == "10.0.0-dev")
+        #expect(plugin.active == true)
+        #expect(result.featureValue == nil)
+    }
+
+    @Test func loadWooCommercePluginAndPOSFeatureSwitch_returns_nil_plugin_and_nil_feature_when_woocommerce_plugin_not_found() async throws {
+        // Given
+        network.simulateResponse(requestUrlSuffix: "system_status", filename: "system-status-wc-plugin-missing")
+
+        // When
+        let result = try await sut.loadWooCommercePluginAndPOSFeatureSwitch(siteID: sampleSiteID)
+
+        // Then
+        #expect(result.wcPlugin == nil)
+        #expect(result.featureValue == nil)
+    }
```

Linked to the comment above regarding not finding the WooCommerce path at this point, is this a expected scenario? Or should we throw an error?

---

Responded in https://github.com/woocommerce/woocommerce-ios/pull/15906#discussion_r2205067059.